gecoscc/userdb.py

```diff
@@ -166,8 +186,52 @@ def get_user(request):
 
     return None
 
-
 def get_groups(userid, request):
     ## TODO
     # Not Implemented
     return ['logged']
+
+def get_ldap_info():
+    settings = get_current_registry().settings
+    ldap = {
+        'url' : settings.get('gecos.ldap.url'),
+        'port': settings.get('gecos.ldap.port'),
+        'prfx': settings.get('gecos.ldap.prfx'),
+        'base': settings.get('gecos.ldap.base'),
+        'admn': settings.get('gecos.ldap.admn'),
+        'pass': settings.get('gecos.ldap.pass')
+    }
+    return ldap
+
+def get_ldap_auth(user,password):
+    ldap_info = get_ldap_info()
+    conn = str(ldap_info['url'])+':'+str(ldap_info['port'])
+    basedn = str(ldap_info['prfx'])+user+","+str(ldap_info['base'])
+
+    try:
+        ld = ldap.initialize(conn)
+        ld.simple_bind_s(basedn,password)
+        result = { 'username': user, 'dn':ld.whoami_s() }
+        ld.unbind_s()
+        return result
+    except ldap.LDAPError, e:
+        ld.unbind_s()
+        return False
+
+def get_ldap_userin(username):
+    ldap_info = get_ldap_info()
+    conn = str(ldap_info['url'])+':'+str(ldap_info['port'])
+    basedn = str(ldap_info['base'])
+    admn = str(ldap_info['admn'])
+    password = str(ldap_info['pass'])
+    sfilter = "(|(%s %s)(sAMAccountName=%s))" % (ldap_info['prfx'], username, username)
```

Comment:
I am not sure every LDAP has got this attribute (sAMAccountName). I think is typical in AD, but not in LDAP.